client/ayon_maya/plugins/load/load_layout.py

```diff
@@ -0,0 +1,223 @@
+from maya import cmds
+import math
+import json
+import collections
+import ayon_api
+from ayon_maya.api import plugin
+from ayon_maya.api.lib import unique_namespace
+from ayon_core.pipeline import (
+    load_container,
+    get_representation_path,
+    discover_loader_plugins,
+    loaders_from_representation,
+    get_current_project_name
+)
+from maya.api import OpenMaya as om
+from ayon_maya.api.pipeline import containerise
+
+
+class LayoutLoader(plugin.Loader):
+    """Layout Loader (json)"""
+
+    product_types = {"layout"}
+    representations = {"json"}
+
+    label = "Layout Loader"
+    order = -10
+    icon = "code-fork"
+    color = "orange"
+
+    def _get_repre_entities_by_version_id(self, data):
+        version_ids = {
+            element.get("version")
+            for element in data
+        }
+        version_ids.discard(None)
+
+        output = collections.defaultdict(list)
+        if not version_ids:
+            return output
+
+        project_name = get_current_project_name()
+        repre_entities = ayon_api.get_representations(
+            project_name,
+            representation_names={"fbx", "abc"},
+            version_ids=version_ids,
+            fields={"id", "versionId", "name"}
+        )
+        for repre_entity in repre_entities:
+            version_id = repre_entity["versionId"]
+            output[version_id].append(repre_entity)
+        return output
+
+    @staticmethod
+    def _get_loader(loaders, product_type):
+        name = ""
+        if product_type in {
+            "rig", "model", "camera",
+            "animation", "staticMesh",
+            "skeletalMesh"}:
+                name = "ReferenceLoader"
+
+        if name == "":
+            return None
+
+        for loader in loaders:
+            if loader.__name__ == name:
+                return loader
+
+        return None
+
+    def get_asset(self, containers, instance_name):
+        asset = None
+        containers = [con for con in containers if con.startswith(instance_name)]
+        for container in containers:
+            namespace = cmds.getAttr(f"{container}.namespace")
+            asset = [asset for asset in cmds.ls(f"{namespace}:*", assemblies=True)][0]
```

Comment:
If there's one thing I've learned from Maya is that even the namespaces are hard to rely on :') hehe.

Anyway, this code is the same as:
```suggestion
        for container in containers:
            namespace = cmds.getAttr(f"{container}.namespace")
            asset = next(iter(cmds.ls(f"{namespace}:*", assemblies=True)))
```

However that would fail as soon as someone had "reparented" the object in their scene locally and it is not 'top level assembly anymore'.

We might want to get the highest root from the container. Also, why do we need to iterate the containers (when would the first container fail?) If there're edge cases there it may be good to document them as comments in the code to explain the reasoning.

Getting the highest root of the container may be someting like this:
```suggestion
        # Get the highest root node from the loaded container
        for container in containers:
            members = lib.get_container_members(container)
            roots = lib.get_highest_in_hierarchy(members)
            root = next(iter(roots), None)
            if root is not None:
                return root
```